### PR TITLE
Elasticsearch: Rename public/app/plugins/datasource/elasticsearch/index_pattern.ts to follow convention

### LINKED ISSUE #55383: Elasticsearch: Rename public/app/plugins/datasource/elasticsearch/index_pattern.ts to follow convention
**Level: easy**

Some files that are maintained by logs squad have snake_case name. According to convention https://github.com/grafana/grafana/blob/765b995b1bcafe8b94d0bbe53737ea6c751d63c9/contribute/style-guides/frontend.md#file-and-directory-naming-conventions we should use:
- `camelCase` when the primary export is a function
- `PascalCase` when the primary export is a class or React component. 

Rename following files  in `grafana/grafana` to follow the above-mentioned convention:
- `public/app/plugins/datasource/elasticsearch/index_pattern.ts`
- `public/app/plugins/datasource/elasticsearch/specs/index_pattern.test.ts` - this should be also moved out of specs folder

### PR DESCRIPTION
**Pull request by mvsousa**
**What this PR does / why we need it**:

Rename public/app/plugins/datasource/elasticsearch/index_pattern.ts to follow convention

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/55383

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/ngalert/eval/evaluator_mock.go [hunk 1]
```diff
@@ -51,10 +51,10 @@ type FakeEvaluator_ConditionEval_Call struct {
 }
 
 // ConditionEval is a helper method to define mock.On call
```
Comment (matyax) [inline pkg/services/ngalert/eval/evaluator_mock.go:53]: This change is out of scope with the issue.

**Review by matyax — CHANGES_REQUESTED** [review]
There's a modified file out of scope of the issue.

File: pkg/services/ngalert/eval/evaluator_mock.go (hunk 1 above)
Comment (mvsousa) [reply pkg/services/ngalert/eval/evaluator_mock.go]: @matyax 
done, removed!
thanks for the review!

Comment (matyax) [reply pkg/services/ngalert/eval/evaluator_mock.go]: Thank you! 

If you don't mind, I'd suggest you to try shorter and more descriptive commit messages. For example:
- `Rename file to follow conventions` or `Elasticsearch: rename file to follow conventions`
- `Revert unwanted change` or `Revert <previous commit name>`.

**Review by matyax — APPROVED** [review]

**Comment by matyax** [comment]
Thank you for your contribution! 🇧🇷
